File: 3336-water-bottles-ii/3336-water-bottles-ii.py
class Solution:
    def maxBottlesDrunk(self, numBottles: int, numExchange: int) -> int:
        
        rem = 0
        Empty = 0
        drink = 0
        while  numExchange <= Empty or numBottles !=0:

            if numExchange > Empty:
                Empty  += numBottles
                drink += numBottles
                numBottles = 0
            elif numExchange <= Empty:
                
                temp = Empty-numExchange
                numExchange += 1
                numBottles +=1
                Empty = temp

        return drink 

File: 3336-water-bottles-ii/3336-water-bottles-ii.java
public class Solution {
    public int maxBottlesDrunk(int numBottles, int numExchange) {
        int rem = 0;
        int Empty = 0;
        int drink = 0;

        while (numExchange <= Empty || numBottles != 0) {
            if (numExchange > Empty) {
                Empty += numBottles;
                drink += numBottles;
                numBottles = 0;
            } else if (numExchange <= Empty) {
                int temp = Empty - numExchange;
                numExchange++;
                numBottles++;
                Empty = temp;
            }
        }

        return drink;
    }
}